rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -79,11 +99,71 @@ public BakeRecipe buildBakeRecipe(
     command.add("kustomize");
     command.add("build");
     command.add(templatePath.getParent().toString());
+
+    BakeRecipe result = new BakeRecipe();
     result.setCommand(command);
+    return result;
+  }
+
+  private BakeRecipe buildBakeRecipeFromGitRepo(
+      BakeManifestEnvironment env, KustomizeBakeManifestRequest request, Artifact artifact)
+      throws IOException {
+    // This is a redundant check for now, but it's here for when we soon remove the old logic of
+    // building from
+    // a github/file artifact type and instead, only support the git/repo artifact type
+    if (!"git/repo".equals(artifact.getType())) {
+      throw new IllegalArgumentException("The inputArtifact should be of type \"git/repo\".");
+    }
+
+    String kustomizeFilePath = request.getKustomizeFilePath();
+    if (kustomizeFilePath == null) {
+      throw new IllegalArgumentException("The bake request should contain a kustomize file path.");
+    }
+
+    Path tarPath = env.resolvePath("repo.tar.gz");
+    try {
+      artifactDownloader.downloadArtifact(artifact, tarPath);
+    } catch (IOException e) {
+      throw new IOException("Failed to download git/repo artifact: " + e.getMessage(), e);
+    }
 
+    try {
+      extractArtifact(tarPath, env.resolvePath(""));
```

Comment:
Currently, we're doing the following:
* Requesting the artifact from clouddriver, then writing that `InputStream` to a file in `artifactDownloader.downloadArtifact()`
* Opening the file as an `InputStream` and decompressing the archive

This would be simpler (and likely more efficient) if we directly consumed the `InputStream` we got back from clouddriver and decrypted from that, rather than needing the intermediate write to disk.

Perhaps we could rename `ArtifactDownloader.downloadArtifact` to `ArtifactDownloader.downloadArtifactToFile` and then have `ArtifactDownloader.downloadArtifact` just return an `InputStream`.

Reply:
Good catch--changed! But I'm not sure how we want to handle an empty response body in the downloadArtifact() method. Right now, I just have it logging an error and returning null.

Reply:
Good question!  I'd suggest throwing an `IOException` (with the text you're currently logging) if the response body is empty.  I realize that's a change from what we currently do, where we just silently don't write out the artifact to a file if there's no input, but it seems like an improvement as the error would be surfaced earlier and more clearly.  (As opposed to currently where it would just manifest as a missing file somewhere downstream.)

On the topic of errors, @maggieneterval just improved the error handling in this file in #453 so that errors actually propagate back to the user (per the screen shots on that PR).  So you'll probably have a minor merge conflict when you merge in her changes here.

Reply:
Sounds good--updated.